### PR TITLE
Fix `make check` on OS X

### PR DESCRIPTION
**Pull request by zbeekman**
Resolve issues with `make check` on OS X

### CODE REVIEW

**Comment by zbeekman** [comment]
I'm gonna give @milancurcic a few hours to say "LGTM" or  comment on required changes, before merging.

**Comment by milancurcic** [comment]
LGTM!! :smile:

**Comment by zbeekman** [comment]
OK, I'm going to double check that the test program will exit non-zero on failure, and then if not, add `stop 1` or something similar, because that releases me of the burden of parsing the output for the Homebrew formula test. If  there's already a `stop n` statement upon failure (with 0 < n < 255) then I'll mint a new release and submit my homebrew PR which I'm ready to pull the trigger on.

**Comment by milancurcic** [comment]
@zbeekman Take a look at the `report_tests` subroutine in `src/tests/datetime_tests`. This subroutine returns to a parent subroutine which returns to the end of the program right away. So I think we need this:

```fortran
SUBROUTINE report_tests(tests)

  LOGICAL,DIMENSION(:),INTENT(IN) :: tests

  INTEGER :: n,ntests,nsuccess,nfailure

  ntests = SIZE(tests)

  nsuccess = 0
  nfailure = 0
  DO n = 1,ntests
    IF(tests(n))THEN
      nsuccess = nsuccess + 1
    ELSE
      nfailure = nfailure + 1
    ENDIF
  ENDDO

  WRITE(UNIT=STDOUT,FMT='(A,I3,A)')'Ran a total of ',ntests,' tests.'
  WRITE(UNIT=STDOUT,FMT='(I3,A,I3,A)')nsuccess,' tests PASSED, ',nfailure,' tests FAILED.'

  ! >>> Suggested code to add <<<
  IF(nfailure > 0)THEN
    STOP 1
  ELSE
    STOP 0
  ENDIF

ENDSUBROUTINE report_tests
```

What do you think?

**Comment by milancurcic** [comment]
I may have not been clear. There is currently no `stop` statements in the test program. I suggest we add this snippet:

```fortran
  IF(nfailure > 0)THEN
    STOP 1
  ELSE
    STOP 0
  ENDIF
```

**Comment by zbeekman** [comment]
I did something similar, but less likely to break backwards compatibility: I added an optional logical argument to that procedure and then call stop in the main test subroutine.

File: src/tests/datetime_tests.f90 [hunk 1]
```diff
@@ -1124,8 +1130,11 @@ SUBROUTINE test_datetime
   !---------------------------------------------------------------------
   WRITE(UNIT=STDOUT,FMT='(71("-"))')
 
-  CALL report_tests(tests)
+  test_failed = .FALSE.
+
+  CALL report_tests(tests,test_failed)
 
+  IF ( test_failed ) STOP 1
```
Comment (zbeekman) [inline src/tests/datetime_tests.f90]: This is the main diff hunk that accomplishes it... we could do it your way too... but in the event that someone reuses `report_tests(...)` I'd rather provide an optional logical argument, to keep the procedure pure, etc.

**Comment by zbeekman** [comment]
@milancurcic Is what I have OK, or do you prefer your way?

**Comment by milancurcic** [comment]
Yours is perfect!

**Comment by zbeekman** [comment]
I'm going to mint a release and then open the Homebrew PR :beers:

**Comment by milancurcic** [comment]
:tada: :balloon: Awesome, it's beer time! :beers:

**Comment by zbeekman** [comment]
haha, well, :beers: is a homebrew logo of sorts... but yes, I'm having a lame Saturday night... well I'm having fun, but most would think it lame.

**Comment by zbeekman** [comment]
Feel free to edit the release notes, @milancurcic...

**Pull request merged**